extensions/wikia/TemplateClassification/TemplateClassificationController.class.php

```diff
@@ -49,4 +54,76 @@ public function getTemplateClassificationEditForm() {
 
 		$this->setVal( 'templateTypes', $templateTypes );
 	}
+
+	/**
+	 * Classify all templates with given category
+	 *
+	 * @requestParam string category category name
+	 * @requestParam string type template type name
+	 *
+	 * @throws ApiException
+	 * @throws \BadRequestApiException
+	 * @throws \InvalidParameterApiException
+	 * @throws \MWException
+	 * @throws \UnauthorizedException
+	 */
+	public function classifyTemplateByCategory() {
+		global $wgCityId;
+
+		$errors = [];
+		$category = $this->request->getVal( 'category', null );
+		$templateType = $this->request->getVal( 'type', null );
+		$userId = $this->wg->User->getId();
+
+		$this->validateRequestForBulkEdit( $this->wg->User, $this->request, $category, $templateType );
+
+		$templates = ( new Helper() )->getTemplatesByCategory( $category );
+		$utcs = new \UserTemplateClassificationService();
+
+		foreach ( $templates as $templateId => $templateTitle ) {
```

Comment:
Some categories could have a lot of templates in them, how many would we want to handle in the request given there is a separate request to the service for each classification?